.github/workflows/pull_request.yml

```diff
@@ -0,0 +1,33 @@
+name: pull-request
+
+on:
+  pull_request:
+    types: [opened, reopened, synchronize]
+
+concurrency: 
+  group: ${{ github.head_ref }}
+  cancel-in-progress: true
```

Comment:
this ensures that if a given PR already is running actions and a new commit is pushed the previous execution is canceled.